packages/synapse-react-client/src/components/AccessRequirementList/AccessRequirementList.tsx

```diff
@@ -232,6 +250,25 @@ export default function AccessRequirementList(
       },
     )
 
+  const sortedGroupedAcessRequirementsByType = useMemo(() => {
+    if (groupedAccessRequirementsByType && sortedAccessRequirementIds) {
```

Comment:
If the grouping `useMemo` turns into effectively a one-liner, I think you could just move the logic here.

```suggestion
    if (accessRequirements && sortedAccessRequirementIds) {
      const groupedAccessRequirementsByType = groupBy(accessRequirements, 'concreteType')
```